### PR TITLE
Merge trait and all related changes

### PR DESCRIPTION
**Pull request by ustulation**
[Review on Reviewable](https://reviewable.io/reviews/dirvine/maidsafe_vault/58)

### CODE REVIEW

File: src/version_handler/mod.rs [hunk 1]
```diff
@@ -19,10 +19,65 @@
...
+
+    fn serialised_contents(&self) -> Vec<u8> {
+        self.data.clone()
+    }
+
+    fn refresh(&self) -> bool {
+        true
+    }
+
+    fn merge(&self, responses: Vec<Box<Sendable>>) -> Option<Box<Sendable>> {
```
Comment (krishnaIndia) [inline src/version_handler/mod.rs]: Not sure about the implementation logic as such for VersionHandler Merge.
But based on David's input on slack, took a median of the vector collected and sending it back. Couldn't get a c++ reference to port

Comment (dirvine) [inline src/version_handler/mod.rs]: Looks fine, we will add a plethora of tests here away so good work man.

**Comment by dirvine** [comment]
Reviewed files:
- Cargo.toml @ r9
- src/data_manager/database.rs @ r9
- src/data_manager/mod.rs @ r9
- src/maid_manager/database.rs @ r9
- src/maid_manager/mod.rs @ r2
- src/main.rs @ r1
- src/pmid_manager/database.rs @ r9
- src/pmid_manager/mod.rs @ r2
- src/utils.rs @ r1
- src/vault.rs @ r2
- src/version_handler/mod.rs @ r9

---





Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/maidsafe_vault/58)

**Comment by dirvine** [comment]
Nice Spandan

---

Reviewed files:
- Cargo.toml @ r10
- src/vault.rs @ r10

---





Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/maidsafe_vault/58)

**Pull request merged**